web_monitoring/db.py

```diff
@@ -657,3 +657,29 @@ def get_version_by_versionista_id(self, versionista_id):
         # Make result look like the result of `get_version` rather than the
         # result of `list_versions`.
         return {'data': result['data'][0]}
+
+    def get_db_uuid_by_versionista_id(self, versionista_id):
```

Comment:
I don’t think this method makes much sense to add. What meaningful utility does it offer over just calling:

```py
client.get_version_by_versionista_id(12345)['data']['uuid']?
```

Reply:
@danielballan wanted this to be added, as we had discussed on slack.

Reply:
Ok, then at the very least, this method shouldn’t duplicate the implementation of `get_version_by_versionista_id()`. It should leverage it. Is there any reason this whole method shouldn’t just be:

```py
def get_db_uuid_by_versionista_id(self, versionista_id):
    """docstring"""
    return self.get_version_by_versionista_id(versionista_id)['data']['id']
```

?

Reply:
I'm not sure about the purpose of adding this method either. @danielballan Is there any specific reason behind this addition?

Reply:
I agree that we don't really need this now. The day before I dove into #114, as I was still catching on the state of things, I was asking whether we already had an easy way to translate from a Versionista `version_id` to our Version UUID. Now, having finished #114, I think `get_version_by_versionista_id` is sufficiently convenient.

In [my comment above](https://github.com/edgi-govdata-archiving/web-monitoring-processing/pull/113#issuecomment-335478685) I meant to redirect your effort to other convenience methods that might add more utility than this one.

Sorry for creating confusion here. I was still catching up!

Reply:
Yeah, the db API hadn't been updated in a while. I guess it's best to add other convenience functions. @danielballan  do you have any in mind, other than the ones you've already mentioned?

Reply:
On second thought, `get_first_version_after(page_id, date)` is just `get_versions(page_id=page_id, start_date=date, chunk_size=1)['data'][0]` which doesn't seem worth a separate function. But `get_previous` still seems useful. Nothing else occurs to me yet.

Reply:
Hmmmm, that query would still have versions in descending date order, though, right? So it would be a little more complex (but the solution to that is probably to make order a parameter in the REST API rather than make a complex algorithm here).

`get_previous(version_id)` would just be:

```py
previous_url = get_version(version_id)['links']['previous']
return previous_url and requests.get(previous_url).json or None

# --- or ---
previous_url = get_version(version_id)['links']['previous']
return previous_url and get_version(previous_url.rsplit('/', 1)[1]) or None
```

So maybe complex enough for a method, maybe not.

Reply:
FYI, edgi-govdata-archiving/web-monitoring-db#164 and edgi-govdata-archiving/web-monitoring-db#165

Reply:
…and it looks like someone showed up out of the blue and is already working on those. So yeah, that’ll turn a bunch more things into one-liners, I guess.

Reply:
@Mr0grog I'm not sure if returning both the previous url and its version content is necessary.
The response of the previous url will have its version id and it can be used to create its url (though I don't see why it is needed because we already have the response).
Also, it is really a matter of usefulness, I suppose. How often would one try to get the previous or the next version? If it'll be used quite frequently, we should definitely add this method (even if it is a one-liner).

Reply:
Yeah, I agree we don't really know yet how frequently certain patterns will be used. I feel confident that the current master branch provides a decent API (not to say optimal but decent) to the db. At this moment, until we know more, it's probably more urgent and productive to focus on the differs. But if you feel inspired to embellish this API, that would be fine too.

Reply:
Yeah, I agree we don't really know yet how frequently certain patterns will be used. I feel confident that the current master branch provides a decent API (not to say optimal but decent) to the db. At this moment, until we know more, it's probably more urgent and productive to focus on the differs. But if you feel inspired to embellish this API, that would be fine too.